### ISSUE TITLE
Official browser support

### ISSUE THREAD

**Issue by alexturpin** [autoNumeric/autoNumeric#539]
I'd like to suggest the establishing of an official list of supported browser versions. I looked around the repo and I can't seem to find anything like it. This would be useful when deciding whether to choose to use the latest version of AutoNumeric, as its use of Map and WeakMap isn't compatible with older versions of IE and Safari.

**Comment by AlexandreBonneau** [comment]
That's a good idea indeed, specially since `4.1` should be released soon™ :)

**Comment by alexturpin** [comment]
A few more notes gathered from the recent deployment of autoNumeric v4 to our userbase:

- As said above, `Map` and `WeakMap` requirements will require the following minimum versions: Safari 8+ and Internet Explorer 11
- [A bug in Safari with `Object.getOwnPropertyDescriptor`](https://bugs.webkit.org/show_bug.cgi?id=140575) means that Safari 9 doesn't work either. I haven't yet figured out what the earliest bug free version is as I can't test Safari myself unfortunately, but if I do find the answer I will report back

**Comment by AlexandreBonneau** [comment]
Thanks for the research @alexturpin 👍

**Comment by evromalarkey** [comment]
Any plan to support IE 9.0+ ?

I've been able to run it on IE 11 with String.prototype.includes polyfill, but in IE 10 and bellow there is more errors.

``Invalid operand in: An object was expected. `` on line 1251
https://github.com/autoNumeric/autoNumeric/blob/2aa57cb9141440a6b632e76bcfe9482426ccf819/src/AutoNumeric.js#L1251

**Comment by AlexandreBonneau** [comment]
Oh my, this thread is making me ashamed for using 'modern' JS functions :sweat_smile: :wink:

About supporting IE9, according to caniuse.com, its marketshare is 0.12% globally, so I'd prefer not having to modify/complexify the code to support it.
Perhaps the best solution would indeed to find polyfills for those new features?

We could then list those in a table in the readme, would that be ok for you?

**Comment by alexturpin** [comment]
I don't think there's any shame in using modern JS functions! That being said, I would love to see a proper list of officially supported browsers. It might also be easy to support IE9+ without much work using something like https://babeljs.io/docs/usage/polyfill/ as part of the build process.

**Comment by AlexandreBonneau** [comment]
Wouldn't babel-polyfill add more weight to the library?
My understanding is that it is required so that webpack would bundle it with AutoNumeric.

If that's the case, I would be against using that, the library being already big enough.
I would prefer to give a CDN link to those few users that still use those old browsers, so that they could load the polyfills separately, if that's possible.

**Comment by alexturpin** [comment]
It would definitely add a bunch of weight to the library! If you'd rather not do that, then I think a more sensible option is listing the supported browsers for the "naked" library as well as what browsers are supported with x functions polyfilled.

**Comment by evromalarkey** [comment]
Yeah loading polyfills separately would be great. I don't like support IE neither, but sometimes it's needed.
